web/src/user_group_edit.js

```diff
@@ -588,11 +588,18 @@ export function change_state(section) {
             group_list_toggler.goto("your-groups");
         } else {
             show_right_section();
-            // We show the list of user groups in the left panel
-            // based on the tab that is active. It is `your-groups`
-            // tab by default.
             redraw_user_group_list();
             switch_to_group_row(group);
+
+            if (left_side_tab === undefined) {
+                left_side_tab = "all-groups";
+                if (user_groups.is_direct_member_of(current_user.user_id, group_id)) {
+                    left_side_tab = "your-groups";
+                }
+            }
+
+            // NOTE: Normally, this will update browser_history. But here we always have a group being edited, so its safe to call.
+            group_list_toggler.goto(left_side_tab);
```

Comment:
Ideally i'd want to change `tabname` inside `redraw_user_group_list` to avoid rewriting the previously drawn list here. However, i need to call `group_list_toggler.goto()` anyway to change the active tab in widget (`redraw_left_panel` in `redraw_user_group_list` simply populates the list, selected tab is managed by `group_list_toggler`). So i've put this to make it clear whats going on.

Reply:
Can you just remove the `redraw_user_group_list()` call above as I guess that would be called when calling `goto`.

Reply:
I have removed the call from here. `stream_settings_ui` has some comments stating redraws being needed for legacy reasons. So still calling `goto` after `redraw_left_panel` there.